packages/app/src/components/approve/ApproveToken.tsx

```diff
@@ -0,0 +1,64 @@
+import { FC, MouseEventHandler } from "react";
+import { Button, Container, Tooltip, OverlayTrigger, Spinner } from "react-bootstrap";
+import { BigNumberish, constants, Contract } from "ethers";
+import { useContractFunction } from "@usedapp/core";
+
+type Props = {
+  symbol: string;
+  hasAllowance: boolean;
+  spender: string;
+  tokenAddress: Contract;
+  amount?: BigNumberish;
+};
+
+const ApproveToken: FC<Props> = ({ symbol, spender, hasAllowance, tokenAddress, amount }) => {
+  const { state: approveTx, send: approveToken } = useContractFunction(tokenAddress, "approve");
+
+  const handleApproval: MouseEventHandler<HTMLDivElement> = async (e) => {
+    e.preventDefault();
+
+    if (!hasAllowance) {
+      await approveToken(spender, amount ?? constants.MaxUint256);
```

Comment:
There is a difference between the Coalesce Nullish operator `??` and the OR operator `||` 

`??` only checks for explicit null/undefined but not what JS considers other 'falsy' values 

It can be written as 

```
if (thisThing != null && 
    thisThing != undefined)
```

While `||` will check for all falsy values and can be written as 

```
if (thisThing != 0 && 
    thisThing != undefined && 
    thisThing != false && 
    thisThing != '' && 
    thisThing != null && 
    thisThing != NaN && 
    thisThing != 0n) 
```
    
    
I believe we want to use `||` here instead

Reply:
yes, we are handling undefined here, I'm aware of the difference